src/diagnostics/OpenPMDWriter.cpp

```diff
@@ -20,29 +20,44 @@ OpenPMDWriter::OpenPMDWriter ()
 }
 
 void
-OpenPMDWriter::InitDiagnostics (const int output_step, const int output_period, const int max_step)
+OpenPMDWriter::InitDiagnostics (const int output_step, const int output_period, const int max_step,
+                                const int nlev)
 {
     HIPACE_PROFILE("OpenPMDWriter::InitDiagnostics()");
 
     // Dump every m_output_period steps and after last step
     if (output_period < 0 ||
        (!(output_step == max_step) && output_step % output_period != 0)) return;
 
-     // pick first available backend if default is chosen
-     if( m_openpmd_backend == "default" ) {
+    // pick first available backend if default is chosen
+    if( m_openpmd_backend == "default" ) {
 #if openPMD_HAVE_HDF5==1
         m_openpmd_backend = "h5";
 #elif openPMD_HAVE_ADIOS2==1
         m_openpmd_backend = "bp";
 #else
         m_openpmd_backend = "json";
 #endif
-     }
+    }
+
+    if (nlev > 1) {
+        for (int lev=0; lev<nlev; ++lev) {
+            std::string filename = m_file_prefix + "/lev_" + std::to_string(lev) +  "/openpmd_%06T."
+                                   + m_openpmd_backend;
+
+            m_outputSeries.push_back(std::make_unique< openPMD::Series >(
+                filename, openPMD::Access::CREATE) );
+            m_last_output_dumped.push_back(-1);
+        }
+    } else {
+        std::string filename = m_file_prefix + "/openpmd_%06T." + m_openpmd_backend;
+
+        m_outputSeries.push_back(std::make_unique< openPMD::Series >(
+            filename, openPMD::Access::CREATE) );
+        m_last_output_dumped.push_back(-1);
+    }
```

Comment:
It may not be very convenient for the user to change the directory structure when adding MR. Could we instead name them `openpmd_%06T.` (level 0, as before) and `openpmd_lev1_%06T.` etc.?

Reply:
I did this initially, but the openPMD viewer reads per directory and then it got confused about different files for the same iteration. Maybe this can be solved with the upcoming add-ons in openPMD. But I think with the current openPMD viewer, this is not useful. @ax3l please correct me, if I am wrong here.

Reply:
thanks for the explanation